src/plugins/flexibleLayout/components/dropHint.vue

```diff
@@ -51,8 +56,25 @@ export default {
             this.isMouseOver = false;
         },
         dropHandler(event) {
-            this.$emit('object-drop-to', event, this.index);
+            this.$emit('object-drop-to', this.index, event);
+            this.isValidTarget = false;
+        },
+        dragstart(event) {
+            if (this.allowDrop && this.allowDrop(event, this.index)) {
```

Comment:
Make the `allowDrop` property required (so it fails loudly) instead of silently failing when it's undefined.